src/Algebra/Consequences/Base.agda

```diff
@@ -13,14 +13,48 @@ module Algebra.Consequences.Base
 open import Algebra.Core
 open import Algebra.Definitions
 open import Data.Sum.Base
+open import Function.Base using (const)
 open import Relation.Binary.Core
 open import Relation.Binary.Definitions using (Reflexive)
+open import Relation.Nullary.Decidable.Core using (yes; no)
+open import Relation.Nullary.Negation.Core using (contradiction)
+open import Relation.Unary using (Pred; Decidable)
+
 
 module _ {ℓ} {_•_ : Op₂ A} (_≈_ : Rel A ℓ) where
 
   sel⇒idem : Selective _≈_ _•_ → Idempotent _≈_ _•_
   sel⇒idem sel x = reduce (sel x x)
 
+module _ {p ℓ} {_•_ : Op₂ A} (_≈_ : Rel A ℓ) (P : Pred A p) where
+
+  almost⇒exceptˡ : _-AlmostLeftCancellative_ _≈_ P _•_ →
+                   Except_-LeftCancellative_ _≈_ P _•_
+  almost⇒exceptˡ cancelˡ {x} with cancelˡ x
+  ... | inj₁ px     = contradiction px
+  ... | inj₂ cancel = const cancel
+
+  almost⇒exceptʳ : _-AlmostRightCancellative_ _≈_ P _•_ →
+                   Except_-RightCancellative_ _≈_ P _•_
+  almost⇒exceptʳ cancelʳ {x} with cancelʳ x
+  ... | inj₁ px     = contradiction px
+  ... | inj₂ cancel = const cancel
+
+module _ {p ℓ} {_•_ : Op₂ A} (_≈_ : Rel A ℓ)
+         {P : Pred A p} (dec : Decidable P) where
+
+  except⇒almostˡ : Except_-LeftCancellative_ _≈_ P _•_ →
+                   _-AlmostLeftCancellative_ _≈_ P _•_
+  except⇒almostˡ cancelˡ x with dec x
+  ... | yes px = inj₁ px
+  ... | no ¬px = inj₂ (cancelˡ ¬px)
+
+  except⇒almostʳ : Except_-RightCancellative_ _≈_ P _•_ →
+                   _-AlmostRightCancellative_ _≈_ P _•_
+  except⇒almostʳ cancelʳ x with dec x
+  ... | yes px = inj₁ px
+  ... | no ¬px = inj₂ (cancelʳ ¬px)
```

Comment:
Indeed, I couldn't be bothered to make these lazier as per my previous strictures against `yes` and `no`... sigh ;-)